src/Components/Address/NewAddressForm.tsx

```diff
@@ -0,0 +1,261 @@
+import * as Yup from "yup"
+import { Button, Column, GridColumns, Input, Message } from "@artsy/palette"
+import { Formik, Form, FormikConfig, useFormikContext } from "formik"
+import { FC, useEffect } from "react"
+import { CountrySelect } from "Components/CountrySelect"
+
+export interface AddressFormValues {
+  name?: string
+  country?: string
+  addressLine1?: string
+  city?: string
+  region?: string
+  postalCode?: string
+  addressLine2?: string
+}
+
+export type AddressChangeHandler = (
+  address: AddressFormValues,
+  key: keyof AddressFormValues
+) => void
+
+export const EMPTY_ADDRESS = {
+  name: "",
+  country: "",
+  addressLine1: "",
+  addressLine2: "",
+  city: "",
+  postalCode: "",
+  region: "",
+}
+
+export const DEFAULT_VALIDATION_SCHEMA = Yup.object().shape({
+  name: Yup.string().required("Name is required"),
+  country: Yup.string().required("Country is required"),
+  addressLine1: Yup.string().required("Address is required"),
+  city: Yup.string().required("City is required"),
+  region: Yup.string().required("Region is required"),
+  postalCode: Yup.string().required("Postal Code is required"),
+})
+
+// For debugging
+export const BasicAddressForm: FC<{
+  onChange?: (args: any) => void
+}> = props => {
+  const { values, errors, touched } = useFormikContext<AddressFormValues>()
+
+  const serializedState = JSON.stringify({ values, errors, touched })
+  // hack to inspect/debug intermediate form state before submission
+  useEffect(() => {
+    props.onChange && props.onChange({ values, errors, touched })
+
+    // really props.onChange should be wrapped in useCallback before being passed in - then we would have this guarantee
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [serializedState])
+  return (
+    <Form>
+      <AddressInputs
+        endContent={
+          <Button type="submit" variant="primaryBlack" width="50%">
+            Save and Continue
+          </Button>
+        }
+      />
+    </Form>
+  )
+}
```

Comment:
Is the debug component necessary? I don't see us wiring up helper components like this often. And a dev can just console.log the formik state in a one-liner from the context or whatever.

Reply:
yeah, this can go away.